app/services/lock-event/service.go

```diff
@@ -0,0 +1,210 @@
+	lock_event "github.com/limechain/hedera-eth-bridge-validator/app/model/lock-event"
+	"github.com/limechain/hedera-eth-bridge-validator/app/model/transfer"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity/fee"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	log "github.com/sirupsen/logrus"
+	"strconv"
+)
+
+// TODO: Create LockEvent Service
+
+type Service struct {
+	bridgeAccount      hedera.AccountID
+	feeRepository      repository.Fee
+	repository         repository.LockEvent
+	distributorService service.Distributor
+	feeService         service.Fee
+	scheduledService   service.Scheduled
+	logger             *log.Entry
+}
+
+func NewService(
+	bridgeAccount string,
+	repository repository.LockEvent,
+	feeRepository repository.Fee,
+	distributor service.Distributor,
+	scheduled service.Scheduled,
+	feeService service.Fee) *Service {
+
+	bridgeAcc, err := hedera.AccountIDFromString(bridgeAccount)
+	if err != nil {
+		log.Fatalf("Invalid bridge account: [%s].", bridgeAccount)
+	}
+
+	return &Service{
+		bridgeAccount:      bridgeAcc,
+		feeRepository:      feeRepository,
+		repository:         repository,
+		distributorService: distributor,
+		feeService:         feeService,
+		scheduledService:   scheduled,
+		logger:             config.GetLoggerFor("Lock Event Service"),
+	}
+}
+
+func (s Service) ProcessEvent(event lock_event.LockEvent) {
+	// TODO: Use proper parameters
+	err := s.repository.Create(event.Id, event.Amount, event.Recipient.String())
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to create a lock event record. Error [%s].", event.Id, err)
+		return
+	}
+
+	// TODO: Prepare Scheduled Token Mint Transaction
+
+	_, feeAmount, transfers, err := s.prepareTransfers(event)
+	if err != nil {
+		s.logger.Errorf("[%s] - Failed to prepare transfers. Error [%s].", event.Id, err)
+		return
+	}
+
+	onExecutionSuccess, onExecutionFail := s.scheduledTxExecutionCallbacks(event.Id, strconv.FormatInt(feeAmount, 10))
+	onSuccess, onFail := s.scheduledTxMinedCallbacks(event.Id)
+
+	s.scheduledService.ExecuteScheduledTransferTransaction(event.Id, event.NativeAsset, transfers, onExecutionSuccess, onExecutionFail, onSuccess, onFail)
+}
+func (s *Service) prepareTransfers(event lock_event.LockEvent) (recipientAmount int64, feeAmount int64, transfers []transfer.Hedera, err error) {
+	fee, remainder := s.feeService.CalculateFee(event.Amount)
+
+	validFee := s.distributorService.ValidAmount(fee)
+	if validFee != fee {
+		remainder += fee - validFee
+	}
+
+	transfers, err = s.distributorService.CalculateMemberDistribution(validFee)
+	if err != nil {
+		return 0, 0, nil, err
+	}
+
+	transfers = append(transfers,
+		transfer.Hedera{
+			AccountID: event.Recipient,
+			Amount:    remainder,
+		},
+		transfer.Hedera{
+			AccountID: s.bridgeAccount,
+			Amount:    -event.Amount,
+		})
+
```

Comment:
These will need to be changed to only an array of two elements (-amount & amount)